include/eve/traits/common_value.hpp

```diff
@@ -8,20 +8,39 @@
 #pragma once
 
 #include <eve/concept/value.hpp>
+#include <eve/concept/simd.hpp>
 #include <eve/traits/as_wide.hpp>
 #include <type_traits>
 
 namespace eve::detail
 {
+  template<typename T>
+  struct find_type_impl {
+    using type = T;
+
+    template<value U>
+    friend consteval find_type_impl<std::conditional_t<
+        scalar_value<T> && scalar_value<U>,
+          decltype(std::declval<T>() + std::declval<U>()),
+          std::conditional_t<
+              (scalar_value<T> && arithmetic_simd_value<U>) ||
+              (arithmetic_simd_value<T> && scalar_value<U>) ||
+              (arithmetic_simd_value<T> && std::same_as<T, U>),
+                std::conditional_t<arithmetic_simd_value<T>, T, U>,
+                void
+          >
```

Comment:
Why would you do this.

Use auto and if constexpr, this is impossible to read.

Reply:
We need sfinae there or we ends up with hard error in some places with ugly messages. That is why the old.code was decltype based.

Reply:
And why thebprevious iteration was overload based.

Reply:
This passes unittests.
I really don't like that it's not symetrical.

```
  template<typename T>
  struct find_common_value_wide_reducer;

  template<>
  struct find_common_value_wide_reducer<void> {};

  template<typename T>
  struct find_common_value_wide_reducer {
    using type = T;

    template <typename U>
    friend auto operator%(
      find_common_value_wide_reducer x,
      find_common_value_wide_reducer<U> y
    ) {
      if constexpr (scalar_value<T> && scalar_value<U>) {
        // Or use decltype if you want
        return find_common_value_wide_reducer<decltype(T{} + U{})>{};
      } else if constexpr ( scalar_value<T> && arithmetic_simd_value<U> ) {
        return y;
      } else if constexpr ( arithmetic_simd_value<T> && (scalar_value<U> || arithmetic_simd_value<U>) ) {
        return x;
      } else {
        return find_common_value_wide_reducer<void>{};
      }
    }
  };

  template<typename... Ts>
  using find_common_value_wide = typename decltype((find_common_value_wide_reducer<Ts>{} % ...))::type;

  template<typename... Ts>
  requires(!scalar_value<Ts> || ... )
  auto find_common_value() -> find_common_value_wide<Ts... >;

```

Reply:
This looks ok for me

Reply:
Since you answered about scalar, answer here - why is left prefred to right? Why between `wide<int8_t>` and `wide<int16_t>` we chose the first one?

Reply:
I see - now it makes sense, sorry - was confused.